### PR TITLE
Create a nodejs-based test case for the dart transpiler.

### PR DESCRIPTION
**Pull request by alexeagle**
This will be useful to repro https://github.com/angular/angular/issues/509 among other issues.
In this change, the actual issue isn't fixed. It only adds the fixture.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ var jsserve = require('./tools/build/jsserve');
 var pubserve = require('./tools/build/pubserve');
 var rundartpackage = require('./tools/build/rundartpackage');
 var karma = require('karma').server;
+var mocha = require('gulp-mocha');
```
Comment (tbosch) [inline gulpfile.js]: Please add `chai` to get `expect` assertions.

**Comment by tbosch** [comment]
Please rewrite existing tests under `docs/**/*.spec.js` into mocha / chai as well.

File: gulpfile.js (hunk 1 above)
Comment (alexeagle) [inline gulpfile.js]: switched to jasmine, now using expect assertions

**Comment by alexeagle** [comment]
dgeni uses jasmine's createSpyObj to do mocking
dgeni/lib/mocks/log.js:2:17
so switching to mocha would also require switching to a different mocking library. That's possible, but feels like more changes than I meant to have for this PR.

How about we stick with Jasmine everywhere, and consider switching to Mocha later?

**Pull request merged**